app/services/zip_delivery_service.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+# Posts zips to S3, if needed.
+class ZipDeliveryService
```

Comment:
if this class doesn't warrant its own spec file, would it make sense to 
- put class in abstract_delivery_job.rb file?
- decompose it into methods in abstract_delivery_job?

Reply:
It probably should have its own spec file broken out of the S3West job's yucky spec, but I didn't care enough to include lots of testing changes in this refactoring.

Reply:
yeah, I'm uncomfortable with that.

Reply:
Of course, totally happy to write up a ticket. Or even just do the work immediately after.